File: Q1 - Two Sum.cpp
/*  Given an array of integers 'nums' and integer 'target', return (in
	any order) the indices of the two numbers such that they add up to the 
	target. Each input has one solution, so no duplicate returns. */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
	
		// Create empty array
        vector<int> returnArr;
        
		// Iterate with 2D array, adding each index i and j
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 0; j < nums.size(); j++) {
			
				// If conditions met, add elements to return array
                if ((nums[i] + nums[j] == target) && (i != j)){
                    returnArr.push_back(i);
                    returnArr.push_back(j);
                }
            }    
        }
        
		// Remove last half of return array, as those are duplicates resulting from above loops
        returnArr.erase(returnArr.begin() + returnArr.size()/2, returnArr.end());
        
        return returnArr;
    }
};

File: Q1 - Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        returnList = []
        for i in range(len(nums)):
            for j in range(len(nums)):
                if (nums[i] + nums[j] == target) and (i != j):
                    if i not in returnList:
                        returnList.append(i)
                    if j not in returnList:
                        returnList.append(j)
        
        return returnList
